sdk/java/runtime/main.go

```diff
@@ -0,0 +1,312 @@
+			return ctr
+		}
+
+		absPath := func(rel ...string) string {
+			return filepath.Join(append([]string{m.moduleConfig.modulePath()}, rel...)...)
+		}
+
+		ctr = ctr.
+			// Get the files from the template
+			WithDirectory(
+				m.moduleConfig.modulePath(),
+				dag.CurrentModule().Source().Directory("template"),
+				dagger.ContainerWithDirectoryOpts{Include: []string{
+					"pom.xml",
+					"**/*.java",
+				}}).
+			// And rename everything so that they match the dagger module name
+			WithExec([]string{"sed", "-i", "-e", fmt.Sprintf("s/dagger-module/%s/g", snakeName), absPath("pom.xml")}).
+			WithExec([]string{"sed", "-i", "-e", fmt.Sprintf("s/DaggerModule/%s/g", camelName), absPath("src", "main", "java", "io", "dagger", "sample", "module", "DaggerModule.java")}).
+			WithExec([]string{"sed", "-i", "-e", fmt.Sprintf("s/dagger-module-name/%s/g", name), absPath("src", "main", "java", "io", "dagger", "sample", "module", "DaggerModule.java")}).
+			WithExec([]string{"sed", "-i", "-e", fmt.Sprintf("s/DaggerModule/%s/g", camelName), absPath("src", "main", "java", "io", "dagger", "sample", "module", "package-info.java")}).
+			WithExec([]string{"sed", "-i", "-e", fmt.Sprintf("s/dagger-module-name/%s/g", name), absPath("src", "main", "java", "io", "dagger", "sample", "module", "package-info.java")}).
+			WithExec([]string{"mv", absPath("src", "main", "java", "io", "dagger", "sample", "module", "DaggerModule.java"), absPath("src", "main", "java", "io", "dagger", "sample", "module", fmt.Sprintf("%s.java", camelName))})
+
+		return ctr
+	}
+}
+
+// generateCode builds and returns the generated source code and java classes
+func (m *JavaSdk) generateCode(
+	ctx context.Context,
+	ctr *dagger.Container,
+	introspectionJSON *dagger.File,
+) *dagger.Directory {
+	// generate the java sdk dependencies
+	javaDeps := m.buildJavaDependencies(ctx, introspectionJSON)
+	// generate the entrypoint class based on the user module
+	entrypoint := ctr.WithExec([]string{"mvn", "clean", "compile"})
+	return dag.
+		Directory().
+		// copy all user files
+		WithDirectory(
+			m.moduleConfig.modulePath(),
+			ctr.Directory(m.moduleConfig.modulePath())).
+		// copy the generated entrypoint under target/generated-sources/annotations
+		WithDirectory(
+			filepath.Join(m.moduleConfig.modulePath(), "target", "generated-sources", "entrypoint"),
+			entrypoint.Directory(filepath.Join(m.moduleConfig.modulePath(), "target", "generated-sources", "annotations"))).
+		// copy the sdk source code under target/generated-sources/sdk
+		// this is not really generated-sources, this is the sdk. But we don't want it as the user source code
+		// and we don't want to install it on the user machine. That way the java classes are made available
+		// to a build system or an IDE without to interfere with the user source code
+		WithDirectory(
+			filepath.Join(m.moduleConfig.modulePath(), "target", "generated-sources", "sdk"),
```

Comment:
This makes sense but can you please find another name instead of "sdk"? The reason is that term is too much overloaded. See:
- https://github.com/dagger/dagger/issues/8424

It's also proven to be harder when filtering out files because it conflicts with other `sdk` dirs, including https://github.com/dagger/dagger/tree/main/sdk. Ideally this would be the name of the package that is published to a public registry. Like `src/dagger` or `vendor/dagger` (generally speaking).

Reply:
I think I covered all places where there was `sdk`, should be better now